tests/load-tests/run-max-concurrency.sh

```diff
@@ -73,14 +73,20 @@ else
         index=$(printf "%04d" "$t")
         cp -vf "$output_dir/load-tests.json" "$output_dir/load-tests.max-concurrency.$index.json"
         cp -vf "$output_dir/load-tests.log" "$output_dir/load-tests.max-concurrency.$index.log"
-        pipelineRunThresholdExceeded=$(jq -rc ".runPipelineSucceededTimeMax > $threshold" "$output_dir/load-tests.json")
-        pipelineRunKPI=$(jq -rc ".runPipelineSucceededTimeMax" "$output_dir/load-tests.json")
-        if [ "$pipelineRunThresholdExceeded" = "true" ]; then
-            echo "The maximal time a pipeline run took to succeed (${pipelineRunKPI}s) has exceeded a threshold of ${threshold}s with $t threads."
+        workloadKPI=$(jq '.workloadKPI' "$output_dir/load-tests.json")
+        if awk "BEGIN { exit !($workloadKPI > $threshold)}"; then
+            echo "The average time a workload took to succeed (${workloadKPI}s) has exceeded a threshold of ${threshold}s with $t threads."
+            indexOld=$(printf "%04d" "$(($t - 1))")
```

Comment:
@jhutar FYI, the "previous" iteration is not the one with `number of threads - 1`. The thread index of the previous iteration is the previous item in the `maxConcurrencySteps` array then one currently stored in variable `t`.

For example: If you have the list of threads:`1,25,50,100` and the current value of `t` is 50, then the previous iteration (the `indexOld`) is 25, not 49.

The test does not run for 49 threads, therefore there are no artifacts captured with that index (`indexOld=0049`) and so the `$output_dir/load-tests.max-concurrency.0049.json` does not exist and the script fails.

Reply:
Should be workarounded. Let me re-run in CI.